quinn-proto/src/connection/streams.rs

```diff
@@ -416,18 +426,23 @@ impl Streams {
 
     /// Cease accepting data on a stream
     ///
-    /// Returns `true` iff the peer should be notified.
-    pub fn stop(&mut self, id: StreamId) -> Result<bool, UnknownStream> {
+    /// Returns a structure which indicates whether this action
+    /// requires transmitting any frames.
+    pub fn stop(&mut self, id: StreamId) -> Result<StopResult, UnknownStream> {
         let stream = match self.recv.get_mut(&id) {
             Some(s) => s,
             None => return Err(UnknownStream { _private: () }),
         };
         stream.assembler.stop();
+        let transmit_stop_sending = ShouldTransmit::new(!stream.is_finished());
```

Comment:
Nit: let's call this `stop_sending` and `transmit_max_data` -> `max_data` so that the initialization in the return value is more streamlined.